runtime/job-workers/server-functions.cpp

```diff
@@ -0,0 +1,73 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2021 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include "runtime/job-workers/shared-memory-manager.h"
+#include "runtime/instance-copy-processor.h"
+
+#include "runtime/job-workers/server-functions.h"
+
+struct {
+  job_workers::SharedMemorySlice *request_memory{nullptr};
+  const char *(*send_reply) (job_workers::SharedMemorySlice *) {nullptr};
+} static current_job;
+
+void init_job_server_interface_lib(job_workers::SharedMemorySlice *request_memory, const char *(*send_reply) (job_workers::SharedMemorySlice *)) noexcept {
+  php_assert(!current_job.request_memory);
+  php_assert(!current_job.send_reply);
+
+  current_job.request_memory = request_memory;
+  current_job.send_reply = send_reply;
+}
+
+void free_job_server_interface_lib() noexcept {
+  if (current_job.request_memory) {
+    vk::singleton<job_workers::SharedMemoryManager>::get().release_slice(current_job.request_memory);
+  }
+  current_job.request_memory = nullptr;
+  current_job.send_reply = nullptr;
+}
+
+class_instance<C$KphpJobWorkerRequest> f$kphp_job_worker_fetch_request() noexcept {
```

Comment:
Now it's possible to fetch the same job request several times. It's not consistent with `fetch` semantic in other places. Maybe it worth to forbid it somehow?

Reply:
It make sense, let's leave as is for a while, and add to TODO.